lib/msal-common/src/utils/Constants.ts

```diff
@@ -213,12 +213,19 @@ export const CodeChallengeMethodValues = {
 };
 
 /**
- * allowed values for response_mode
+ * allowed values for server response type
  */
-export const ResponseMode = {
+export const ServerResponseType = {
     QUERY: "query",
     FRAGMENT: "fragment",
-    FORM_POST: "form_post",
+} as const;
+export type ServerResponseType = typeof ServerResponseType[keyof typeof ServerResponseType];
+
+/**
+ * allowed values for response_mode
+ */
+export const ResponseMode = {
+    ...ServerResponseType, FORM_POST: "form_post"
```

Comment:
Let's make sure we have the business logic to back "form_post" up as an option. If we only have successful code paths for `query` and `hash` then we should reconsider allowing the config option.

Reply:
Could you clarify what you mean by this?

Reply:
We don't have any code that actually processes a response in "form_post" format do we? If that's the case we probably don't need to include it as an option here.

Reply:
Oh, this is the old ResponseMode type. One of the earlier comments said to avoid repetition in the types, and since I had trouble removing form_post from ResponseMode to make ServerResponseType, I did it the other way around. form_post is there to make it the same as ResponseMode was before.

Reply:
Ah, that's great then.